python/pyspark/sql/types.py

```diff
@@ -2384,6 +2408,14 @@ def verify_struct(obj: Any) -> None:
 
         verify_value = verify_struct
 
+    elif isinstance(dataType, VariantType):
+
+        def verify_variant(obj: Any) -> None:
+            # The variant data type can take in any type.
+            pass
+
+        verify_value = verify_variant
```

Comment:
Hm, don't we need `createDataFrame`, `collect` and Python UDF test cases that uses this type? If you plan to do them in separate PRs, that's fine.

Reply:
Yeah, I was on the fence for adding these tests because we need to add additional python support for variant expressions (i.e. `parse_json` which turns strings into variant data types). Happy to add this support then add these test cases